atlasdb-feign/src/main/java/com/palantir/atlasdb/http/AtlasDbErrorDecoder.java

```diff
@@ -27,6 +27,8 @@
 import feign.codec.ErrorDecoder;
 
 public class AtlasDbErrorDecoder implements ErrorDecoder {
+    private static final String LOCK_METHOD_KEY = "RemoteLockService#lock(String,LockRequest)";
```

Comment:
this would also make testing easier as we can just write a unit test on `AwaitingLeadershipProxy` given that we don't currently have a good mechanism for forcing a node to step down as leader (https://github.com/palantir/atlasdb/issues/2100)